File: python/minscheme/stdenv.py
from minscheme import builtins
from minscheme import data


def make_top_level_env() -> data.Environment:
    env = data.Environment(parent=None)
    for name, func in builtins.ALL_MAP.items():
        env.ensure(name).value = data.Func(name, func)
    return env

File: typescript/minscheme/stdenv.ts
import {Environment} from "./data";
import {allBuiltins} from "./builtins";

export function createTopLevelEnv(): Environment {
  const env = new Environment();
  for (const [name, value] of allBuiltins) {
    env.ensure(name).value = value;
  }
  return env;
}
